app/cmd/replica.go

```diff
@@ -116,18 +118,14 @@ func startReplica(c *cli.Context) error {
 		}
 	}
 
-	// The --volume-name flag for the replica command (added before the identity validation feature) and the
-	// --volume-name global flag are redundant. Since both are optional, proceed with either one, but do not proceed if
-	// they are different.
-	volumeName, err := resolveVolumeName(c.String("volume-name"), c.GlobalString("volume-name"))
-	if err != nil {
-		return err
-	}
+	volumeName := c.GlobalString("volume-name")
 	replicaInstanceName := c.String("replica-instance-name")
 	dataServerProtocol := c.String("data-server-protocol")
 
 	controlAddress, dataAddress, syncAddress, syncPort, err :=
-		util.GetAddresses(volumeName, address, types.DataServerProtocol(dataServerProtocol))
+		// TODO (5845): Remove replica command-specific volume-name flag when its use is completely replaced in
+		// longhorn-manager.
+		util.GetAddresses(c.String("volume-name"), address, types.DataServerProtocol(dataServerProtocol))
```

Comment:
Why will you use `c.String("volume-name")` rather than variable `volumeName` here?

Reply:
My intention is for this to be temporary. Until longhorn-manager changes eventually merge, longhorn-manager will not be aware of the global flag on master-head and will continue to provide the replica command specific volume-name flag. This might cause annoying test failures, etc. I am thinking we can merge this longhorn-engine PR, allowing me to merge the longhorn-instance-manager and longhorn-manager PRs. Then I can come back and completely remove the command-specific flag per the code comments.

Reply:
For this kind of change that involves multiple repos and compatibility issues, we will merge all PRs at the same time. There is no need to worry about temporary mismatchings between the engine PR merge and the manager PR merge. So please directly use the global flags in this PR.

Reply:
I currently import the longhorn-engine commits on my own branch in my longhorn-instance-manager and longhorn-manager PRs. This allows me to test those changes, since the commits aren't in the longhorn/longhorn-engine repo yet.  My intention was to change each to reference the correct. My intention was to fix these imports after the longhorn-engine changes merged and before the longhorn-instance-manager and longhorn-manager changes were approved. Please let me know what the best practice here is.

Reply:
Agree with @shuo-wu about the using the global flags in this PR directly. 

In general, since `longhorn-engine` is imported by `longhorn-manager` and `longhorn-instance-manager`, I think we can use this flow:
1. Update the `longhorn-engine` PR repo as if  `longhorn-manager` and `longhorn-instance-manager` are already updated
2. In  `longhorn-manager` and `longhorn-instance-manager` PRs, use the replace directive in `go.mod` to point to the in-progress engine PR
3. When all `longhorn-engine`, `longhorn-manager`, `longhorn-instance-manager` are approved and ready for merging. 
    1. `longhorn-engine` PR will be merged first 
    2. In  `longhorn-manager` and `longhorn-instance-manager` PRs, edit `go.mod` to point to the correct engine commits 
    3. Merger  `longhorn-manager` and `longhorn-instance-manager` PRs
    4. This PRs merging process should happen without a few hours so there is no need to worry about temporary missmatchings